leakcanary-deobfuscation-gradle-plugin/src/main/java/com/squareup/leakcanary/deobfuscation/CopyObfuscationMappingFileTask.kt

```diff
@@ -0,0 +1,54 @@
+package com.squareup.leakcanary.deobfuscation
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+@CacheableTask
+open class CopyObfuscationMappingFileTask : DefaultTask() {
+
+  @Input
+  var variantDirName: String? = null
+
+  @Input
+  @SkipWhenEmpty
+  @PathSensitive(PathSensitivity.RELATIVE)
+  var mappingFile: File? = null
```

Comment:
use your empty file here as the default value. This properties are var, they will be overriden by devs using the DSL. If not, then the file is empty by default and the task is skipped.
Oh, but I realize that no, we want to be less defensive here, and more assertive: we are in the case where the variant has been filtered by devs, there should be a mapping file, if not, we crash. That's it. We don't want to skip the task: if the task is setup by the plugin, then it should run.